crates/bevy_asset/src/handle.rs

```diff
@@ -149,26 +155,34 @@ impl<T: Asset> Handle<T> {
         }
     }
 
-    /// Get a copy of this handle as a Weak handle
-    pub fn as_weak<U: Asset>(&self) -> Handle<U> {
```

Comment:
Could we keep around a safer version of this API too / instead?

Reply:
How so? Make it return a Result?

Reply:
#3522 might give you an ideia of why. As François mentioned its actual use case is to reuse the asset id for `Font` and `FontAtlasSet` assets, [here](https://github.com/bevyengine/bevy/blob/main/crates/bevy_text/src/glyph_brush.rs#L99) and [here](https://github.com/bevyengine/bevy/blob/main/examples/ui/font_atlas_debug.rs#L39), which is why I proposed a rename. At least until we figure out a better way to link fonts and font atlases as cart said. `clone_weak()` is the actual method one would use to create a weak copy.

Reply:
Ugh, right. Okay cool, I'm happy with this change then.